csm/core/services/usl.py

```diff
@@ -149,273 +146,104 @@ async def _format_lyve_pilot_access_params(
-                'secretKey': iam_user_access_key.secret_key,
+                'accessKey': iam_user_access_key.get('access_key_id'),
+                'secretKey': iam_user_access_key.get('secret_key'),
             },
         }
         return access_params
 
-    async def _cleanup_on_lyve_pilot_registration_error(
-        self,
-        s3_account_service: S3AccountService,
-        s3_account: Dict,
-        iam_user: IamUser,
-        iam_user_access_key: IamUserCredentials,
-        bucket: Bucket,
-    ) -> None:
-        Log.debug('Cleaning up Lyve Pilot resources...')
-        access_key = s3_account.get('access_key')
-        secret_key = s3_account.get('secret_key')
-        Log.debug('Remove bucket')
-        s3_client = self._s3plugin.get_s3_client(
-            access_key, secret_key, CsmS3ConfigurationFactory.get_s3_connection_config()
-        )
-        await s3_client.delete_bucket(bucket.name)
-        Log.debug('Remove IAM user access key')
-        iam_client = self._s3plugin.get_iam_client(
-            access_key, secret_key, CsmS3ConfigurationFactory.get_iam_connection_config()
-        )
-        await iam_client.delete_user_access_key(
-            iam_user_access_key.access_key_id, user_name=iam_user.user_name)
-        Log.debug('Remove IAM user')
-        await iam_client.delete_user(iam_user.user_name)
-        Log.debug('Remove S3 account')
-        s3_credentials = S3Credentials(
-            str(s3_account.get('account_name')),
-            str(s3_account.get('access_key')),
-            str(s3_account.get('secret_key')),
-            '',
-        )
+    async def _cleanup(self,
+                       s3_account_service: S3AccountService,
+                       s3_iam_users_service: IamUsersService,
+                       s3_access_keys_service: S3AccessKeysService,
+                       s3_bucket_service: S3BucketService,
+                       s3_credentials: Optional[S3Credentials] = None,
+                       s3_account_name: Optional[str] = None,
+                       iam_user_name: Optional[str] = None,
+                       iam_user_access_key_id: Optional[str] = None,
+                       bucket_name: Optional[str] = None) -> None:
+        if s3_credentials is None:
+            Log.debug('No cleaning to be done')
+            return
         try:
-            await s3_account_service.delete_account(
-                s3_credentials, s3_credentials.user_id
-            )
+            if bucket_name is not None:
+                Log.debug(f'Deleting bucket {bucket_name}')
+                await s3_bucket_service.delete_bucket(bucket_name, s3_credentials)
+            if iam_user_access_key_id is not None:
+                Log.debug(f'Deleting access key {iam_user_access_key_id}')
+                await s3_access_keys_service.delete_access_key(
+                    s3_credentials, iam_user_access_key_id, user_name=iam_user_name)
```

Comment:
`iam_user_name` could be potentially null at this point, at least from a typing perspective.

Reply:
The thing is there is really no need in manually handling IAM user's access key. It's done already inside IAM users service. Just removing these lines.